ddtrace/llmobs/_llmobs.py

```diff
@@ -1387,89 +1395,82 @@ def submit_evaluation(
                                 evaluation metric.
         """
         if cls.enabled is False:
             log.debug(
-                "LLMObs.submit_evaluation() called when LLMObs is not enabled. Evaluation metric data will not be sent."
+                "LLMObs.submit_evaluation() called when LLMObs is not enabled. "
+                "Evaluation metric data will not be sent."
             )
             return
+
         error = None
         try:
-            if not isinstance(span_context, dict):
+            if (
+                not isinstance(span_context, dict)
+                or not isinstance(span_context.get("span_id"), str)
+                or not isinstance(span_context.get("trace_id"), str)
+            ):
                 error = "invalid_span"
-                log.warning(
-                    "span_context must be a dictionary containing both span_id and trace_id keys. "
+                raise TypeError(
+                    "`span_context` must be a dictionary containing both span_id and trace_id keys. "
                     "LLMObs.export_span() can be used to generate this dictionary from a given span."
                 )
-                return
-
-            ml_app = ml_app if ml_app else config._llmobs_ml_app
-            if not ml_app:
-                error = "missing_ml_app"
-                log.warning(
-                    "ML App name is required for sending evaluation metrics. Evaluation metric data will not be sent. "
-                    "Ensure this configuration is set before running your application."
-                )
-                return
 
             timestamp_ms = timestamp_ms if timestamp_ms else int(time.time() * 1000)
 
             if not isinstance(timestamp_ms, int) or timestamp_ms < 0:
                 error = "invalid_timestamp"
-                log.warning("timestamp_ms must be a non-negative integer. Evaluation metric data will not be sent")
-                return
+                raise ValueError("timestamp_ms must be a non-negative integer. Evaluation metric data will not be sent")
 
-            span_id = span_context.get("span_id")
-            trace_id = span_context.get("trace_id")
-            if not (span_id and trace_id):
-                error = "invalid_span"
-                log.warning(
-                    "span_id and trace_id must both be specified for the given evaluation metric to be submitted."
-                )
-                return
             if not label:
                 error = "invalid_metric_label"
-                log.warning("label must be the specified name of the evaluation metric.")
-                return
+                raise ValueError("label must be the specified name of the evaluation metric.")
 
-            if not metric_type or metric_type.lower() not in ("categorical", "numerical", "score"):
+            if not metric_type or metric_type.lower() not in ("categorical", "score", "boolean"):
```

Comment:
Technically we haven't fully dropped support for numerical metrics yet (we've only deprecated and convert them to score metrics atm). Let's avoid raising errors for this until the major version release.